### PR TITLE
Find output files recursively after a doc build

### LINKED ISSUE #3112: Find a way to recursively search for `.nc`/`.jld2` files within the `docs` directory
At the moment we hardcode the locations where output files in docs builds are so we can delete them before deploying the docs. See:

https://github.com/CliMA/Oceananigans.jl/blob/4551a78b1f3fe4bb3b238676c128dc751be9b934/docs/make.jl#L165-L169

We should find a way to automatically search for output files recursively in the `docs` directory and delete them before deploying the docs.

I've asked in Julia slack how to do this and some ideas include:

```Julia
julia> function recursive_find(dir, pattern)
           mapreduce(vcat, walkdir(dir)) do (root, dirs, files)
               joinpath.(root, filter(contains(pattern), files))
           end
       end
recursive_find (generic function with 1 method)

julia> recursive_find(DEPOT_PATH[end], r"\.jl$")
988-element Vector{String}:
 "/opt/julia/julia-1.9.0/share/julia/julia-config.jl"
 "/opt/julia/julia-1.9.0/share/julia/base/Base.jl"
 "/opt/julia/julia-1.9.0/share/julia/base/Enums.jl"
 "/opt/julia/julia-1.9.0/share/julia/base/abstractarray.jl"
 ⋮
 "/opt/julia/julia-1.9.0/share/julia/test/testhelpers/llvmpasses.jl"
 "/opt/julia/julia-1.9.0/share/julia/test/testhelpers/withlocales.jl"
 "/opt/julia/julia-1.9.0/share/julia/test/unicode/utf8.jl"
```

### PR DESCRIPTION
**Pull request by navidcy**
Resolves #3112

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -162,11 +161,23 @@ makedocs(bib,
 
 @info "Cleaning up temporary .jld2 and .nc output created by doctests or literated examples..."
 
-for file in vcat(glob("docs/*.jld2"),
-                 glob("docs/*.nc"),
-                 glob("docs/build/generated/*.jld2"),
-                 glob("docs/build/generated/*.nc")
-                 )
```
Comment (tomchor) [inline docs/make.jl:169]: Honestly I prefer the old code. It's not as elegant a solution as this PR is proposing, but it's way clearer and more concise. I believe it also achieves the same result, no?

**Review by glwagner — APPROVED** [review]

File: docs/make.jl (hunk 1 above)
Comment (glwagner) [reply docs/make.jl:169]: Right. The code is only the same under a very specific assumption.

**Comment by navidcy** [comment]
seems like the docs error on buildkite but locally it works -- strange.

File: docs/make.jl [hunk 2]
```diff
@@ -162,11 +161,23 @@ makedocs(bib,
...
+Return list of filepaths within `directory` that fall under the `pattern::Regex`, e.g., `pattern = r"\.jl"`.
+"""
+recursive_find(directory, pattern) =
+    mapreduce(vcat, walkdir(directory)) do (root, dirs, files)
+        joinpath.(root, filter(contains(pattern), files))
+    end
+
+files = []
+
+for pattern in [r"\.jld2", r"\.nc"]
```
Comment (tomchor) [inline docs/make.jl:176]: The error happens with escaping, so maybe the code below will work (works for me locally)

```suggestion
recursive_find(directory, pattern) =
    mapreduce(vcat, walkdir(directory)) do (root, dirs, files)
        joinpath.(root, filter(endswith(pattern), files))
    end

files = []

for pattern in [r".jld2", r".nc"]
```

Comment (navidcy) [reply docs/make.jl]: Thanks!!

**Comment by navidcy** [comment]
> Same error... no idea what's happening since there are no more scaped characters. Would changing `r".jld2"` to `raw".jld2"` do anything?

error is in line 168 so I'm wondering if it's simply the docstring!

**Comment by navidcy** [comment]
Let's see if this works. If so, I'll bring the examples back. :)

**Pull request merged**